src/loader.js

```diff
@@ -21,7 +21,9 @@ export type I18nLoaderType = {
   ) => {locale: string | Locale, translations: TranslationsObjectType},
 };
 
-const loader: () => I18nLoaderType = () => {
+const defaultResolveLocales = (ctx: Context) => ctx.headers['accept-language'];
+
+const loader: () => I18nLoaderType = (resolveLocales = defaultResolveLocales) => {
```

Comment:
Can we update the explicit type definition for `loader` here?  It is currently `() => I18nLoaderType` but now expects an optional `resolveLocales` argument.

As a starting point, maybe something like:
```
type LoaderFactoryType = (resolver?: (ctx: Context) => string) => I18nLoaderType;
```